easybuild/tools/filetools.py

```diff
@@ -1171,3 +1171,87 @@ def det_size(path):
         _log.warn("Could not determine install size: %s" % err)
 
     return installsize
+
+
+def find_flexlm_license(custom_env_vars=None, lic_specs=None):
+    """
+    Find FlexLM license.
+
+    Considered specified list of environment variables;
+    checks for path to existing license file or valid license server specification.
+
+    If no license is found through environment variables, also consider 'lic_specs'.
+
+    @param custom_env_vars: list of environment variables to considered (if None, only consider $LM_LICENSE_FILE)
+    @param lic_specs: list of license specifications
+    @return: tuple with list of valid license specs found and name of first valid environment variable
+    """
+    valid_lic_specs = []
+    lic_env_var = None
+
+    # regex for license server spec; format: <port>@<server>
+    server_port_regex = re.compile(r'^[0-9]+@\S+$')
+
+    # always consider $LM_LICENSE_FILE
+    lic_env_vars = ['LM_LICENSE_FILE']
+    if isinstance(custom_env_vars, basestring):
+        lic_env_vars.insert(0, custom_env_vars)
+    elif custom_env_vars is not None:
+        lic_env_vars = custom_env_vars + lic_env_vars
+
+    # grab values for defined environment variables
+    cand_lic_specs = {}
+    for env_var in lic_env_vars:
+        if env_var in os.environ:
+            cand_lic_specs[env_var] = os.environ[env_var].split(os.pathsep)
+
+    # also consider provided license spec (last)
+    # use None as key to indicate that these license specs do not have an environment variable associated with them
+    if lic_specs:
+        cand_lic_specs[None] = lic_specs
+
+    _log.debug("Candidate license specs: %s", cand_lic_specs)
+
+    # check for valid license specs
+    # order matters, so loop over original list of environment variables to consider
+    valid_lic_specs = []
+    for env_var in lic_env_vars + [None]:
+        values = cand_lic_specs.get(env_var, None) or []
+        _log.info("Considering %s to find FlexLM license specs: %s", env_var, values)
+
+        for value in values:
+
+            # license files to consider
+            lic_files = None
+            if os.path.isfile(value):
```

Comment:
The function will fail with something like:

```
    if os.path.isfile(value):
  File "/usr/lib64/python2.7/genericpath.py", line 29, in isfile
    st = os.stat(path)
TypeError: coercing to Unicode: need string or buffer, NoneType found
```

if none of the expected environment variables are not defined AND the easyconfig has no `license_file` defined.

The problem is that `values` becomes `[None]`, then `value` is `None`, and `os.path.isfile` breaks.